### PR TITLE
JSUI-2964 Remove DynamicFacet & Facet Dependencies in FieldValue

### PR DESCRIPTION
**Pull request by ThibodeauJF**
FieldValue file went from 13006 lines down to 550 lines
https://coveord.atlassian.net/browse/JSUI-2964

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/ui/Facet/Facet.ts [hunk 1]
```diff
@@ -1026,6 +1027,15 @@ export class Facet extends Component {
     return _.map(this.values.getSelected(), (value: FacetValue) => value.value);
   }
 
+  /**
+   * Checks if a single value is selected.
+   * @param value The name of the facet value to verify.
+   */
+  public hasSelectedValue(value: string) {
+    const facetValue = this.values.get(value);
```
Comment (samisayegh) [inline src/ui/Facet/Facet.ts:1035]: Why not use `this.values.hasSelectedValue` here similar to line 389 in `DynamicFacet.ts`?

Comment (ThibodeauJF) [reply src/ui/Facet/Facet.ts:1035]: It doesn't exist for Facet values. We could always create it though

File: src/ui/DynamicFacet/DynamicFacet.ts [hunk 2]
```diff
@@ -381,6 +381,14 @@ export class DynamicFacet extends Component implements IDynamicFacet {
     this.deselectMultipleValues([value]);
   }
 
+  /**
+   * Checks if a single value is selected.
```
Comment (fbeaudoincoveo) [inline src/ui/DynamicFacet/DynamicFacet.ts:385]: Suggestion:

> Determines whether the specified value is selected in the facet.

File: src/ui/Facet/Facet.ts [hunk 3]
```diff
@@ -1026,6 +1027,15 @@ export class Facet extends Component {
     return _.map(this.values.getSelected(), (value: FacetValue) => value.value);
   }
 
+  /**
+   * Checks if a single value is selected.
```
Comment (fbeaudoincoveo) [inline src/ui/Facet/Facet.ts:1031]: Same suggestion as above :)

**Pull request merged**